expression/builtin_time_vec.go

```diff
@@ -1352,11 +1352,40 @@ func (b *builtinWeekWithoutModeSig) vecEvalInt(input *chunk.Chunk, result *chunk
 }
 
 func (b *builtinUnixTimestampDecSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinUnixTimestampDecSig) vecEvalDecimal(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+
+	timeBuf, err := b.bufAllocator.get(types.ETTimestamp, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(timeBuf)
+	if err := b.args[0].VecEvalTime(b.ctx, input, timeBuf); err != nil {
+		return err
+	}
+	result.ResizeDecimal(n, false)
+	Ts := result.Decimals()
+	for i := 0; i < n; i++ {
+		if timeBuf.IsNull(i) {
+			result.SetNull(i, true)
+			continue
+		}
+		t, err := timeBuf.GetTime(i).Time.GoTime(getTimeZone(b.ctx))
+		if err != nil {
+			Ts[i] = *new(types.MyDecimal)
```

Comment:
We don't need to do this since `ResizeDecimal(n, false)` will set all elements to the zero value of `MyDecimal`.